// @aws-sdk/client-personalize-events/dist-cjs/schemas/schemas_0.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.PutUsers$ = exports.PutItems$ = exports.PutEvents$ = exports.PutActions$ = exports.PutActionInteractions$ = exports.User$ = exports.PutUsersRequest$ = exports.PutItemsRequest$ = exports.PutEventsRequest$ = exports.PutActionsRequest$ = exports.PutActionInteractionsRequest$ = exports.MetricAttribution$ = exports.Item$ = exports.Event$ = exports.ActionInteraction$ = exports.Action$ = exports.errorTypeRegistries = exports.ResourceNotFoundException$ = exports.ResourceInUseException$ = exports.InvalidInputException$ = exports.PersonalizeEventsServiceException$ = void 0;
const _A = "Action";
const _AI = "ActionId";
const _AIL = "ActionInteractionsList";
const _AIc = "ActionInteraction";
const _AIct = "ActionImpression";
const _AL = "ActionList";
const _E = "Event";
const _EL = "EventList";
const _I = "Item";
const _II = "ItemId";
const _IIE = "InvalidInputException";
const _IL = "ItemList";
const _Im = "Impression";
const _MA = "MetricAttribution";
const _PA = "PutActions";
const _PAI = "PutActionInteractions";
const _PAIR = "PutActionInteractionsRequest";
const _PAR = "PutActionsRequest";
const _PE = "PutEvents";
const _PER = "PutEventsRequest";
const _PI = "PutItems";
const _PIR = "PutItemsRequest";
const _PU = "PutUsers";
const _PUR = "PutUsersRequest";
const _RIUE = "ResourceInUseException";
const _RNFE = "ResourceNotFoundException";
const _SJAIP = "SynthesizedJsonActionInteractionProperties";
const _SJAP = "SynthesizedJsonActionProperties";
const _SJEPJSON = "SynthesizedJsonEventPropertiesJSON";
const _SJIP = "SynthesizedJsonItemProperties";
const _SJUP = "SynthesizedJsonUserProperties";
const _U = "User";
const _UI = "UserId";
const _UL = "UserList";
const _a = "application/json";
const _aI = "actionId";
const _aIc = "actionInteractions";
const _ac = "actions";
const _c = "client";
const _dA = "datasetArn";
const _e = "error";
const _eAS = "eventAttributionSource";
const _eI = "eventId";
const _eL = "eventList";
const _eT = "eventType";
const _eV = "eventValue";
const _h = "http";
const _hE = "httpError";
const _i = "impression";
const _iI = "itemId";
const _it = "items";
const _m = "message";
const _mA = "metricAttribution";
const _mT = "mediaType";
const _p = "properties";
const _rI = "recommendationId";
const _s = "smithy.ts.sdk.synthetic.com.amazonaws.personalizeevents";
const _sA = "sentAt";
const _sI = "sessionId";
const _se = "sensitive";
const _t = "timestamp";
const _tI = "trackingId";
const _u = "users";
const _uI = "userId";
const n0 = "com.amazonaws.personalizeevents";
const schema_1 = require("@smithy/core/schema");
const errors_1 = require("../models/errors");
const PersonalizeEventsServiceException_1 = require("../models/PersonalizeEventsServiceException");
const _s_registry = schema_1.TypeRegistry.for(_s);
exports.PersonalizeEventsServiceException$ = [-3, _s, "PersonalizeEventsServiceException", 0, [], []];
_s_registry.registerError(exports.PersonalizeEventsServiceException$, PersonalizeEventsServiceException_1.PersonalizeEventsServiceException);
const n0_registry = schema_1.TypeRegistry.for(n0);
exports.InvalidInputException$ = [-3, n0, _IIE,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
n0_registry.registerError(exports.InvalidInputException$, errors_1.InvalidInputException);
exports.ResourceInUseException$ = [-3, n0, _RIUE,
    { [_e]: _c, [_hE]: 409 },
    [_m],
    [0]
];
n0_registry.registerError(exports.ResourceInUseException$, errors_1.ResourceInUseException);
exports.ResourceNotFoundException$ = [-3, n0, _RNFE,
    { [_e]: _c, [_hE]: 404 },
    [_m],
    [0]
];
n0_registry.registerError(exports.ResourceNotFoundException$, errors_1.ResourceNotFoundException);
exports.errorTypeRegistries = [
    _s_registry,
    n0_registry,
];
var ActionId = [0, n0, _AI, 8, 0];
var ItemId = [0, n0, _II, 8, 0];
var SynthesizedJsonActionInteractionProperties = [0, n0, _SJAIP, { [_mT]: _a, [_se]: 1 }, 0];
var SynthesizedJsonActionProperties = [0, n0, _SJAP, { [_mT]: _a, [_se]: 1 }, 0];
var SynthesizedJsonEventPropertiesJSON = [0, n0, _SJEPJSON, { [_mT]: _a, [_se]: 1 }, 0];
var SynthesizedJsonItemProperties = [0, n0, _SJIP, { [_mT]: _a, [_se]: 1 }, 0];
var SynthesizedJsonUserProperties = [0, n0, _SJUP, { [_mT]: _a, [_se]: 1 }, 0];
var UserId = [0, n0, _UI, 8, 0];
exports.Action$ = [3, n0, _A,
    0,
    [_aI, _p],
    [0, [() => SynthesizedJsonActionProperties, 0]], 1
];
exports.ActionInteraction$ = [3, n0, _AIc,
    0,
    [_aI, _sI, _t, _eT, _uI, _eI, _rI, _i, _p],
    [[() => ActionId, 0], 0, 4, 0, [() => UserId, 0], 0, 0, [() => ActionImpression, 0], [() => SynthesizedJsonActionInteractionProperties, 0]], 4
];
exports.Event$ = [3, n0, _E,
    8,
    [_eT, _sA, _eI, _eV, _iI, _p, _rI, _i, _mA],
    [0, 4, 0, 1, [() => ItemId, 0], [() => SynthesizedJsonEventPropertiesJSON, 0], 0, [() => Impression, 0], () => exports.MetricAttribution$], 2
];
exports.Item$ = [3, n0, _I,
    0,
    [_iI, _p],
    [0, [() => SynthesizedJsonItemProperties, 0]], 1
];
exports.MetricAttribution$ = [3, n0, _MA,
    0,
    [_eAS],
    [0], 1
];
exports.PutActionInteractionsRequest$ = [3, n0, _PAIR,
    0,
    [_tI, _aIc],
    [0, [() => ActionInteractionsList, 0]], 2
];
exports.PutActionsRequest$ = [3, n0, _PAR,
    0,
    [_dA, _ac],
    [0, [() => ActionList, 0]], 2
];
exports.PutEventsRequest$ = [3, n0, _PER,
    0,
    [_tI, _sI, _eL, _uI],
    [0, 0, [() => EventList, 0], [() => UserId, 0]], 3
];
exports.PutItemsRequest$ = [3, n0, _PIR,
    0,
    [_dA, _it],
    [0, [() => ItemList, 0]], 2
];
exports.PutUsersRequest$ = [3, n0, _PUR,
    0,
    [_dA, _u],
    [0, [() => UserList, 0]], 2
];
exports.User$ = [3, n0, _U,
    0,
    [_uI, _p],
    [0, [() => SynthesizedJsonUserProperties, 0]], 1
];
var __Unit = "unit";
var ActionImpression = [1, n0, _AIct,
    0, [() => ActionId,
        0]
];
var ActionInteractionsList = [1, n0, _AIL,
    0, [() => exports.ActionInteraction$,
        0]
];
var ActionList = [1, n0, _AL,
    0, [() => exports.Action$,
        0]
];
var EventList = [1, n0, _EL,
    0, [() => exports.Event$,
        0]
];
var Impression = [1, n0, _Im,
    0, [() => ItemId,
        0]
];
var ItemList = [1, n0, _IL,
    0, [() => exports.Item$,
        0]
];
var UserList = [1, n0, _UL,
    0, [() => exports.User$,
        0]
];
exports.PutActionInteractions$ = [9, n0, _PAI,
    { [_h]: ["POST", "/action-interactions", 200] }, () => exports.PutActionInteractionsRequest$, () => __Unit
];
exports.PutActions$ = [9, n0, _PA,
    { [_h]: ["POST", "/actions", 200] }, () => exports.PutActionsRequest$, () => __Unit
];
exports.PutEvents$ = [9, n0, _PE,
    { [_h]: ["POST", "/events", 200] }, () => exports.PutEventsRequest$, () => __Unit
];
exports.PutItems$ = [9, n0, _PI,
    { [_h]: ["POST", "/items", 200] }, () => exports.PutItemsRequest$, () => __Unit
];
exports.PutUsers$ = [9, n0, _PU,
    { [_h]: ["POST", "/users", 200] }, () => exports.PutUsersRequest$, () => __Unit
];
